template/cordova/lib/MSBuildTools.js

```diff
@@ -44,8 +52,39 @@ MSBuildTools.prototype.buildProject = function(projFile, buildType, buildarch, o
             args.push('/p:' + key + '=' + otherConfigProperties[key]);
         });
     }
+    var that = this;
 
-    return spawn(path.join(this.path, 'msbuild'), [projFile].concat(args), { stdio: 'inherit' });
+    // Check if SDK required to build the respective platform is present. If not present, return with corresponding error, else call msbuild.
+    if (projFile.indexOf('CordovaApp.Phone.jsproj') > -1) {
```

Comment:
These 3 blocks below are almost identical. It could more convenient if `checkPhoneSDK` and `checkWinSDK` would return a promise, rejected with corresponding error message if SDK is not found. Then you could just call `check` method and call `spawn` on success.